eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/json/EclairModels.scala

```diff
@@ -258,16 +263,48 @@ case class PaymentRequest(
     tags: Vector[JsObject],
     signature: LnInvoiceSignature)
 
-sealed abstract class PaymentResult
-case class PaymentSucceeded(
-    amountMsat: MilliSatoshis,
-    paymentHash: Sha256Digest,
-    paymentPreimage: String,
-    route: JsArray)
-    extends PaymentResult
+case class PaymentResult(
+  id: String,
+  paymentHash: Sha256Digest,
+  preimage: Option[PaymentPreimage],
+  amountMsat: MilliSatoshis,
+  createdAt: FiniteDuration,
+  completedAt: Option[FiniteDuration],
+  status: PaymentStatus)
+
+case class ReceivedPaymentResult(
+  paymentHash: Sha256Digest,
+  amountMsat: MilliSatoshis,
+  receivedAt: FiniteDuration)
+
+//object PaymentStatus extends Enumeration {
+//  val PENDING = Value(1, "PENDING")
+//  val SUCCEEDED = Value(2, "SUCCEEDED")
+//  val FAILED = Value(3, "FAILED")
+//}
+
+sealed trait PaymentStatus
+object PaymentStatus {
+  case object PENDING extends PaymentStatus
+  case object SUCCEEDED extends PaymentStatus
+  case object FAILED extends PaymentStatus
+
+  def apply(s: String): PaymentStatus = s match {
+    case "PENDING" => PENDING
+    case "SUCCEEDED" => SUCCEEDED
+    case "FAILED" => FAILED
```

Comment:
Add a catch-all case at the end that gives a better error message than `MatchError`?